lua/nvim-treesitter/configs.lua

```diff
@@ -261,8 +264,12 @@ function M.is_enabled(mod, lang)
     return false
   end
 
-  for _, parser in pairs(module_config.disable) do
-    if lang == parser then return false end
+  if type(module_config.enable) == 'table' then
```

Comment:
the root of the problem is that we use `enable` both as a config variable and a state variable. I guess there is already some logic that assumes that `enable` is a state, so it's difficult to decouple